protobuf/proto-types/mempool.proto

```diff
@@ -39,7 +39,8 @@ message Replaced {
   required int32    replaced_vsize        = 2;  // The vsize of the replaced transaction.
   required int64    replaced_fee          = 3;  // The fee of the replaced transaction.
   required uint64   replaced_entry_time   = 4;  // The entry_time of the replaced transaction.
-  required bytes    replacement_txid      = 5;  // The txid of the replacement transaction.
+  required bytes    replacement_txid      = 5;  // The replacement transaction ID or package hash.
   required int32    replacement_vsize     = 6;  // The vsize of the replacement transaction.
   required int64    replacement_fee       = 7;  // The fee of the replacement transaction.
+  required bool     replaced_by_transaction = 8;  // Boolean indicating if replacement_txid is a transaction ID (true) or package hash (false).
```

Comment:
The `MempoolReplaced` also needs a an update on the ebpf side, the ctypes (used to map the ebpf C struct to a Rust struct), and the mapping of the Rust struct to the protobuf struct.

ebpf struct:
https://github.com/0xB10C/peer-observer/blob/8a49c622e21a14a08f4d40df8f16bfaf5830e874/extractors/ebpf/src/bpf/tracing.bpf.c#L366-L374

ctypes: 
https://github.com/0xB10C/peer-observer/blob/8a49c622e21a14a08f4d40df8f16bfaf5830e874/shared/src/ctypes.rs#L317-L333

mapping of Rust to protobuf (and `Display` implementation):
https://github.com/0xB10C/peer-observer/blob/8a49c622e21a14a08f4d40df8f16bfaf5830e874/shared/src/mempool.rs#L92-L105